src/fight.cpp

```diff
@@ -3921,8 +3921,8 @@ int calculate_vision_penalty(struct char_data *ch, struct char_data *victim) {
   }
 
   // Pre-calculate the things we care about here. First, character vision info.
-  bool ch_has_ultrasound = AFF_FLAGGED(ch, AFF_ULTRASOUND);
-  bool ch_has_thermographic = AFF_FLAGGED(ch, AFF_INFRAVISION) || CURRENT_VISION(ch) == THERMOGRAPHIC;
+  bool ch_has_ultrasound = AFF_FLAGGED(ch, AFF_ULTRASOUND) || has_vision(ch, VISION_ULTRASONIC);
```

Comment:
I want to get rid of AFF_ULTRASOUND here if I can, or at least have AFF_ULTRASOUND consumed behind the scenes to grant VISION_ULTRASONIC. Removing this and the other aff flags from the logic will simplify things.